Card/src/main/java/com/braintreepayments/api/card/CardClient.kt

```diff
@@ -0,0 +1,141 @@
+package com.braintreepayments.api.card
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import com.braintreepayments.api.card.CardNonce.Companion.fromJSON
+import com.braintreepayments.api.core.ApiClient
+import com.braintreepayments.api.core.BraintreeClient
+import com.braintreepayments.api.core.BraintreeException
+import com.braintreepayments.api.core.Configuration
+import com.braintreepayments.api.core.GraphQLConstants
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Used to tokenize credit or debit cards using a [Card]. For more information see the
+ * [documentation](https://developer.paypal.com/braintree/docs/guides/credit-cards/overview)
+ */
+class CardClient {
+
+    private val braintreeClient: BraintreeClient
+    private val apiClient: ApiClient
+
+    /**
+     * Initializes a new [CardClient] instance
+     *
+     * @param context an Android Context
+     * @param authorization a Tokenization Key or Client Token used to authenticate
+     */
+    constructor(context: Context, authorization: String) : this(
+        BraintreeClient(
+            context,
+            authorization
+        )
+    )
+
+    @VisibleForTesting
+    internal constructor(braintreeClient: BraintreeClient) : this(
+        braintreeClient,
+        ApiClient(braintreeClient)
+    )
+
+    @VisibleForTesting
+    internal constructor(
+        braintreeClient: BraintreeClient,
+        apiClient: ApiClient
+    ) {
+        this.braintreeClient = braintreeClient
+        this.apiClient = apiClient
+    }
+
+    /**
+     * Create a [CardNonce].
+     *
+     *
+     * The tokenization result is returned via a [CardTokenizeCallback] callback.
+     *
+     *
+     *
+     * On success, the [CardTokenizeCallback.onCardResult] method will be
+     * invoked with a [CardResult.Success] including a nonce.
+     *
+     *
+     *
+     * If creation fails validation, the [CardTokenizeCallback.onCardResult]
+     * method will be invoked with a [CardResult.Failure] including an
+     * [ErrorWithResponse] exception.
+     *
+     *
+     *
+     * If an error not due to validation (server error, network issue, etc.) occurs, the
+     * [CardTokenizeCallback.onCardResult] method will be invoked with a
+     * [CardResult.Failure] with an [Exception] describing the error.
+     *
+     * @param card     [Card]
+     * @param callback [CardTokenizeCallback]
+     */
+    fun tokenize(card: Card, callback: CardTokenizeCallback) {
+        braintreeClient.sendAnalyticsEvent(CardAnalytics.CARD_TOKENIZE_STARTED)
+        braintreeClient.getConfiguration { configuration: Configuration?, error: Exception? ->
+            if (error != null) {
+                callbackFailure(callback, CardResult.Failure(error))
+                return@getConfiguration
+            }
+            val shouldTokenizeViaGraphQL =
+                configuration!!.isGraphQLFeatureEnabled(
```

Comment:
Can we handle the case when `configuration` is null? 

We return an error in PayPalClient: https://github.com/braintree/braintree_android/blob/f13626a796ae46b21fe41d62b0092c7f73d3c036/PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalClient.kt#L75

Reply:
Yeah it would be awesome if we can remove state from all feature clients. There's potential for `CardClient.configuration` to become stale in an app that hosts multiple merchants e.g. a marketplace app.